### PR TITLE
feat: Add options.rootProperty

### PR DESCRIPTION
**Pull request by raymondfeng**
### Description

1. Drop Node 8.x support
2. Upgrade dependencies
3. Add `options.rootProperty` to control the root property of json/xml response
    - OpenAPI specs can define `error` object in the schema. Such endpoints need to send conforming errors to the client. Always adding `error` as the root property break such cases.

### Checklist

- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,7 +55,7 @@ error handler:
 const http = require('http');
 const writeErrorToResponse = require('strong-error-handler')
   .writeErrorToResponse;
-const errHandlingOptions = {debug: process.env.NODE_ENV === 'development'}
+const errHandlingOptions = { debug: process.env.NODE_ENV === 'development' };
```
Comment (bajtos) [inline README.md:58]: Unrelated change, it's also changing from LoopBack style (no spaces around brackets) to Node.js style.

Please revert.

File: README.md [hunk 2]
```diff
@@ -38,10 +38,12 @@ var app = express();
...
-app.use(errorHandler({
-  debug: app.get('env') === 'development',
-  log: true,
-}));
+app.use(
+  errorHandler({
+    debug: app.get('env') === 'development',
+    log: true,
+  })
+);
```
Comment (bajtos) [inline README.md:46]: Unrelated change, please revert.

File: README.md [hunk 3]
```diff
@@ -196,45 +199,45 @@ To migrate a LoopBack 2.x application to use `strong-error-handler`:
...
-    }
+<pre>
+"final:after": {
+"strong-error-handler": {
+"params": {
+"debug": true,
+"log": true
+}
+}
+}
```
Comment (bajtos) [inline README.md:237]: Unrelated changes, please revert them. 

Also note that you are breaking indentation of the pre-formatted text.

File: README.md [hunk 4]
```diff
@@ -252,17 +255,17 @@ The same error generated when `debug: true` :
...
+  at Context.<anonymous> (User/strong-error-handler/test/handler.test.js:220:21)
+  at callFnAsync (User/strong-error-handler/node_modules/mocha/lib/runnable.js:349:8)
+  at Test.Runnable.run (User/strong-error-handler/node_modules/mocha/lib/runnable.js:301:7)
+  at Runner.runTest (User/strong-error-handler/node_modules/mocha/lib/runner.js:422:10)
+  at User/strong-error-handler/node_modules/mocha/lib/runner.js:528:12
+  at next (User/strong-error-handler/node_modules/mocha/lib/runner.js:342:14)
+  at User/strong-error-handler/node_modules/mocha/lib/runner.js:352:7
+  at next (User/strong-error-handler/node_modules/mocha/lib/runner.js:284:14)
+  at Immediate._onImmediate (User/strong-error-handler/node_modules/mocha/lib/runner.js:320:5)
+  at tryOnImmediate (timers.js:543:15)
```
Comment (bajtos) [inline README.md:268]: Removing trailing space is fine 👍 I would prefer to do that in a standalone commit though.

File: test/handler.test.js [hunk 5]
```diff
@@ -728,6 +754,30 @@ describe('strong-error-handler', function() {
...
 
+    it('honors options.rootProperty', function(done) {
+      const error = new ErrorWithProps({
+        name: 'ValidationError',
+        message: 'The model instance is not valid.',
+        statusCode: 422,
+        details: 'some details',
+        extra: 'sensitive data',
+      });
+      givenErrorHandlerForError(error, {rootProperty: 'myRoot'});
```
Comment (bajtos) [inline test/handler.test.js:765]: What is the expected XML response when `rootProperty` is set to `false`?

File: README.md [hunk 6]
```diff
@@ -100,27 +102,28 @@ The `strong-error-handler` package supports JSON, HTML and XML responses:
...
+| defaultType          | String                    | `"json"`  | Specifies the default response content type to use when the client does not provide any Accepts header.                                                                                                                                                                                                                                                   |
+| rootProperty         | String                    | `"error"` | Specifies the root property name for json or xml                                                                                                                                                                                                                                                                                                          |
```
Comment (bajtos) [inline README.md:119]: Please describe that `false` value is also allowed and explain what will a JSON and a XML response look like when `rootProperty` is set to `false`.

File: lib/send-json.js [hunk 7]
```diff
@@ -7,8 +7,17 @@
...
+module.exports = function sendJson(res, data, options) {
+  options = options || {};
+  let err = data;
+  if (options.rootProperty == null) {
+    err = {error: data};
+  } else if (typeof options.rootProperty === 'string') {
+    if (options.rootProperty) {
+      err = {[options.rootProperty]: data};
+    }
+  }
```
Comment (bajtos) [inline lib/send-json.js:19]: These conditions are very difficult to understand, it's not clear that `rootProperty: false` is intentionally supported.

Proposed solution:

```js
const err = options.rootProperty === false
  ? data
  : {[options.rootProperty || 'error']: data};
```

I think it would be best to extract this bit into a function that can be shared with the XML serializer too.

File: README.md [hunk 8]
```diff
@@ -14,9 +14,9 @@ In debug mode, `strong-error-handler` returns full error stack traces and intern
 
 ## Supported versions
 
-Current|Long Term Support|Maintenance
-:-:|:-:|:-:
-3.x|2.x|1.x
+| Current | Long Term Support | Maintenance |
+| :-----: | :---------------: | :---------: |
+|   3.x   |        2.x        |     1.x     |
```
Comment (bajtos) [inline README.md:19]: Since we are going to publish this PR as `4.0.0`, we need to update this table too. Please use the format we have in the main loopback repository (see https://github.com/strongloop/loopback#module-long-term-support-policy), were each version line has it's own table row and the table has the following columns: Version, Status, Published, EOL

**Comment by raymondfeng** [comment]
@bajtos PTAL.

File: lib/send-html.js [hunk 9]
```diff
@@ -17,7 +17,7 @@ const compiledTemplates = {
 module.exports = sendHtml;
 
 function sendHtml(res, data, options) {
-  const toRender = {options: {}, data: data};
+  const toRender = {options: options, data: data};
```
Comment (bajtos) [inline lib/send-html.js:20]: Nitpick:

```suggestion
  const toRender = {options, data};
```
